src/map/ai/controllers/trust_controller.cpp

```diff
@@ -90,6 +90,7 @@ void CTrustController::DoCombatTick(time_point tick)
 
 void CTrustController::DoRoamTick(time_point tick)
 {
+    // FIXME: POwner->PMaster->PAI can be null. When does this get nulled?
     if (POwner->PMaster->PAI->IsEngaged())
```

Comment:
This needs to be resolved before this can be merged.

Reply:
Fixed in https://github.com/DarkstarProject/darkstar/pull/4739/commits/1106cc023d734f28fe1d94cabe07235161683d51

Reply:
Does the PMaster pointer still exist?  The only situation I can think of off the top of my head where PAI is null is after the entity is destroyed and you're looking at dead memory (which may or may not set the pointer to null).

If the owner is not setting its trusts PMasters to null when they are destroyed/zone out/whatever, that's probably the best place to start

Reply:
It was because I wasn't clearing the trusts from the zone list, so when the mob tried to interact with the trust and this logic executed, it would crash. Fixed now.

Reply:
What would a mob have to do with a trusts roam tick?

I am not sure if deletePET is the right function to call here, but I can't look until I have access to an actual repository and not GitHub's shitty web search

Reply:
Oh yeah, the zone server's ZoneServer will handle removing the pet from the zone list, so you shouldn't have to (it'll go as soon as the trust's status is STATUS_DISAPPEAR, after fadeout animation and stuff).  Removing it too early will stop its AI and I think you will end up having it leak memory because the zone server didn't delete it.  You should be fine just setting its PMaster to nullptr when it leaves the zone (since they have to persist for a few seconds for the despawn animation)

Reply:
or.. just set PMaster to nullptr in Despawn.  That would work too

Reply:
Done. https://github.com/DarkstarProject/darkstar/pull/4739/commits/76ee2a62689cc9dcd26109ca9a628c97e7de60f3

Reply:
sorry, that was my fault for not rereading DeletePET and assuming it still frees the pet from memory (i swear it did at some point)